Sources/Private/CoreAnimation/CoreAnimationLayer.swift

```diff
@@ -98,13 +98,17 @@ final class CoreAnimationLayer: BaseAnimationLayer {
   /// The `AnimationTextProvider` that `TextLayer`'s use to retrieve texts,
   /// that they should use to render their text context
   var textProvider: AnimationTextProvider {
-    didSet { reloadTexts() }
+    didSet {
+      // We need to rebuild the current animation after registering a value provider,
+      // since any existing `CAAnimation`s could now be out of date.
```

Comment:
```suggestion
      // We need to rebuild the current animation after updating the text provider,
      // since this is used in `TextLayer.setupAnimations(context:)`
```

Reply:
[7a71e9a](https://github.com/airbnb/lottie-ios/pull/1723/commits/7a71e9abe94549199da5b36689bed6f7cd338703) ✅